File: jung-algorithms/src/main/java/edu/uci/ics/jung/layout/util/LayoutGraphEvent.java
package edu.uci.ics.jung.layout.util;

import com.google.common.graph.Graph;
import edu.uci.ics.jung.layout.model.Point;

/**
 * A LayoutEvent that also includes a reference to the Graph
 *
 * @param <N>
 */
public class LayoutGraphEvent<N> extends LayoutEvent<N> {

  final Graph<N> graph;

  public LayoutGraphEvent(N node, Graph<N> graph, Point location) {
    super(node, location);
    this.graph = graph;
  }

  public LayoutGraphEvent(LayoutEvent<N> layoutEvent, Graph<N> graph) {
    super(layoutEvent.getNode(), layoutEvent.location);
    this.graph = graph;
  }

  public Graph<N> getGraph() {
    return graph;
  }
}

File: jung-algorithms/src/main/kotlin/edu/uci/ics/jung/layout/util/LayoutGraphEvent.kt
package edu.uci.ics.jung.layout.util

import com.google.common.graph.Graph
import edu.uci.ics.jung.layout.model.Point

/**
 * A LayoutEvent that also includes a reference to the Graph
 *
 * @param N
 */
class LayoutGraphEvent<N : Any> : LayoutEvent<N> {

  val graph: Graph<N>

  constructor(node: N, graph: Graph<N>, location: Point) : super(node, location) {
    this.graph = graph
  }

  constructor(layoutEvent: LayoutEvent<N>, graph: Graph<N>) : super(layoutEvent.node, layoutEvent.location) {
    this.graph = graph
  }
}
